### PR TITLE
Multi-GPUS support

### PR DESCRIPTION
**Pull request by MlWoo**
Many friends seem very to be interested in multi-gpus support when training the model. Maybe it is necessary to merge the branch into the master one.

### CODE REVIEW

**Comment by MlWoo** [comment]
@begeekmyfriend  I have not modified the relative code in terms of the pattern.

**Comment by begeekmyfriend** [comment]
@Rayhane-mamah Yes I agree. In multi-gpu mode we can set `r=1` and expand the batch size to retain smooth gradient. So please consider it as another branch.

**Comment by tomse-h** [comment]
When I try to use this Fork as it is, I run into the following:

_ValueError: Cannot feed value of shape (48, 408, 1025) for Tensor 'datafeeder/linear_targets:0', which has shape '(?, ?, 513)'_

What could be the cause of this? I preprocessed LJSpeech with the given hyperparameters btw.

**Comment by MlWoo** [comment]
@tomse-h I have not modified the relative code in terms of the linear pattern. You can complete it with the solution of mel features

**Comment by shaktikshri** [comment]
I might be a bit late into this conversation, but did you guys also see a proportional increase in sec/step when using multiple GPUs? Here are my stats on V100 GPUs with outputs_per_step = 16
#GPU----batchsize----sec/step
1.................32......................~4
2.................64.....................~10
3.................96 ....................~15
4.................128....................~19

**Comment by MlWoo** [comment]
@shaktikshri No, it increases but does scale linearly. You would better check the time of loading data and the unbalance of length of data of each device.
